src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,43 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    Animal.alive.push(this);
+  }
+  bite(target) {
+    const status = target instanceof Herbivore;
+
+    if (target.hidden === false && status) {
+      target.health -= 50;
+    }
+
+    if (!target.health) {
```

Comment:
Just check if health <= 0
![image](https://user-images.githubusercontent.com/91411692/231816351-94378888-3045-42eb-85cb-6789be8f882c.png)
```suggestion
    if (target.health <= 0) {
```